### PR TITLE
Remove alloca usage [RDY]

### PR DESCRIPTION
**Pull request by aravindprasads**
Fix for issue - https://github.com/radcli/radcli/issues/42 

Removed use of alloca as it is non-POSIX.

### CODE REVIEW

**Comment by aravindprasads** [comment]
Hi @nmav ,

1) Can we remove alloca.h file too ? 

2) If this change looks fine, can we plan to merge it after the dictionary changes merge since the location of changes for alloca conflict with dictionary changes.

File: lib/dict.c [hunk 1]
```diff
@@ -42,7 +42,8 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
 	char            attrstr[AUTH_ID_LEN];
 	char            typestr[AUTH_ID_LEN];
 	char            optstr[AUTH_ID_LEN];
-	char            *cp, *ifilename;
+	char            ifilename[AUTH_ID_LEN] = {0};
```
Comment (nmav) [inline lib/dict.c]: not introduced in this patch but `AUTH_ID_LEN` has nothing to do with limits of filenames and is in fact very small (64). What about fixing it using `PATH_MAX`?

File: lib/dict.c [hunk 2]
```diff
@@ -265,12 +266,11 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
...
 			}
-			ifilename = namestr;
+			strncpy(ifilename, namestr, sizeof(ifilename));
 			/* Append directory if necessary */
 			if (namestr[0] != '/') {
 				cp = strrchr(filename, '/');
 				if (cp != NULL) {
-					ifilename = alloca(AUTH_ID_LEN);
 					*cp = '\0';
 					snprintf(ifilename, AUTH_ID_LEN, "%s/%s", filename, namestr);
```
Comment (nmav) [inline lib/dict.c]: we can use `sizeof(ifilename)` as you do above, to avoid copying around the size.

**Comment by aravindprasads** [comment]
Hi @nmav 

Yes, changed size to PATH_MAX and fixed sizeof(). 

Kindly review and let me know your views. 

If the changes look fine, can we plan to merge it after the dictionary changes merge since the location of changes for alloca conflict with dictionary changes.

**Comment by aravindprasads** [comment]
Hi @nmav 

I have rebased the pull request with latest of Master branch. 

Kindly review the same.

**Comment by nmav** [comment]
Thank you!

**Pull request merged**